aiida/orm/nodes/data/structure.py

```diff
@@ -1213,9 +1213,20 @@ def get_composition(self):
 
         :returns: a dictionary with the composition
         """
+        import numpy as np
         symbols_list = [self.get_kind(s.kind_name).get_symbols_string() for s in self.sites]
-        composition = {symbol: symbols_list.count(symbol) for symbol in set(symbols_list)}
-        return composition
+        gcd = np.gcd(symbols_list.count(symbol) for symbol in set(symbols_list))
+
+        if type == 0:
```

Comment:
As suggested above, using strings instead of integers is preferred since the string gives more of a hint of what it does.
```suggestion
        if mode == 'full':
```